dotnet/src/Microsoft.Agents.AI.Hosting.OpenAI/Responses/Converters/ResponseConverterExtensions.cs

```diff
@@ -0,0 +1,193 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using Microsoft.Agents.AI.Hosting.OpenAI.Responses.Common;
+using Microsoft.Agents.AI.Hosting.OpenAI.Responses.Common.Id;
+using Microsoft.Agents.AI.Hosting.OpenAI.Responses.Invocation;
+using Microsoft.Agents.AI.Hosting.OpenAI.Responses.Models;
+using Microsoft.Extensions.AI;
+
+namespace Microsoft.Agents.AI.Hosting.OpenAI.Responses.Converters;
+
+/// <summary>
+/// Extension methods for converting agent responses to Response models.
+/// </summary>
+internal static class ResponseConverterExtensions
+{
+    /// <summary>
+    /// Converts an AgentRunResponse to a Response model.
+    /// </summary>
+    /// <param name="agentRunResponse">The agent run response to convert.</param>
+    /// <param name="request">The original create response request.</param>
+    /// <param name="context">The agent invocation context.</param>
+    /// <returns>A Response model.</returns>
+    public static Response ToResponse(this AgentRunResponse agentRunResponse, CreateResponse request,
+        AgentInvocationContext context)
+    {
+        var output = agentRunResponse.Messages
+            .SelectMany(msg => msg.ToItemResource(context.IdGenerator, context.JsonSerializerOptions));
+
+        return new Response
+        {
+            Id = context.ResponseId,
+            CreatedAt = (agentRunResponse.CreatedAt ?? DateTimeOffset.UtcNow).ToUnixTimeSeconds(),
+            Model = request.Agent?.Name ?? request.Model ?? "unknown",
+            Status = ResponseStatus.Completed,
+            Agent = request.Agent?.ToAgentId(),
+            Conversation = new ConversationReference { Id = context.ConversationId },
+            Metadata = request.Metadata is IReadOnlyDictionary<string, string> metadata ? new Dictionary<string, string>(metadata) : null,
+            Output = output.ToList(),
+            Usage = agentRunResponse.Usage.ToResponseUsage(),
+            ParallelToolCalls = true,
+            Tools = []
+        };
```

Comment:
Isn't more detail typically roundtripped from the request, settings like temperature and top_p and the like?